test/extended/util/client.go

```diff
@@ -337,6 +345,40 @@ func (c *CLI) SetupProject() string {
 	return newNamespace
 }
 
+func (c *CLI) setupNamespace() string {
+	requiresTestStart()
+	newNamespace := names.SimpleNameGenerator.GenerateName(fmt.Sprintf("e2e-test-%s-", c.kubeFramework.BaseName))
+	c.SetNamespace(newNamespace).ChangeUser(fmt.Sprintf("%s-user", newNamespace))
```

Comment:
ChangeUser requires the users.user API.

Reply:
This was covered here: https://github.com/openshift/origin/pull/27358/files#diff-e21320d17fdce22cad99f368c20f8aca8b28bac96f890d69e95fda9be0ae7f93R822

Even though it means that `UserConfig` == `AdminConfig`

Reply:
>This was covered here: https://github.com/openshift/origin/pull/27358/files#diff-e21320d17fdce22cad99f368c20f8aca8b28bac96f890d69e95fda9be0ae7f93R822

>Even though it means that UserConfig == AdminConfig

That is functionally incorrect.  The reason we switch users is to prove that the features function properly when run as an unprivileged user.  Given differences in admission and REST handling, using an admin user doesn't test the functionality.

Reply:
Agree, was not familiar with the project API. Now the service account has the same permissions as a project user.